.husky/common.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
```

Comment:
Maybe here on in the particular scripts, we should also set some defaults for cases when the variables are not defined. Something like this should work:

```sh
export MP_GIT_HOOKS_ENABLE="${MP_GIT_HOOKS_ENABLE:-true}"
```

As for the defaults, I think we could do more or less these:
```sh
MP_GIT_HOOKS_ENABLE=true
MP_GIT_HOOKS_ESLINT=true
MP_GIT_HOOKS_STYLELINT=true
MP_GIT_HOOKS_PHPLINT=true
MP_GIT_HOOKS_CODE_SNIFFER=true
MP_GIT_HOOKS_MINIMAL_PLUGIN_STANDARDS=true
MP_GIT_HOOKS_PHPSTAN=false
MP_GIT_HOOKS_INSTALL_JS=false
MP_GIT_HOOKS_INSTALL_PHP=false
```

Or do you think we should set different defaults? To me, it seems that these would sort of preserve BC, while disabling PHPStan, that is problematic and slow.

Reply:
I was on the fence about including defaults. I was thinking I would just publish a P2 and everyone could update their own `.env` based on their preferences, but maintaining backwards compatibility makes sense to me.

I added these defaults in 9ac1b84287385aec2e6c97c91eba80d63addd56c. I decided to leave phpstan enabled for true BC, but if you feel strongly about defaulting to `false` I'm ok changing it.

On the topic of PhpStan, I think the command could be improved to optionally only run on staged files (basically make the command accept a list of files to scan), which would speed things up if the cache gets cleared. I started playing around with this but ran into a problem when the robofile itself failed phpstan. I would need to make sure it doesn't run on files we don't normally include in the scan. Does this seem like a worthwhile idea to pursue? I could write a janitor ticket.

Reply:
> On the topic of PhpStan, I think the command could be improved to optionally only run on staged files

@johnolek Is that possible with PHPStan? I think it can't work. Unlike linters, PHPStan can't view files independently, meaning that scanning only some files is likely impossible, or, more precisely, the type inference would then work only from the specific files, leading to incomplete or unknown types, and if you cause type incompatibilities elsewhere, it wouldn't catch them.

It actually seems to be explained here: https://phpstan.org/blog/why-you-should-always-analyse-whole-project

What we could do, though, is to stash all the new unstaged files, and after running the command we could pop the stash.

Reply:
That makes sense, thanks for the explanation and link. I think it's probably best to just leave it alone.